corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -742,6 +775,22 @@ def get_queue_obj_id(queue_obj):
             return getattr(queue_obj, queue_id_param)
         self.get_queue_obj_id = get_queue_obj_id
 
+        def get_cached_list_key():
+            return "partial_queues.queued_or_processing.%s" % run_timestamp
+        self.get_cached_list_key = get_cached_list_key
+
+    def add_processing_doc_id(self, doc_id):
+        client = get_redis_connection()
+        client.rpush(self.get_cached_list_key(), doc_id)
```

Comment:
Does this key ever expire? If not, will it ever be removed if a migration crashes?

Reply:
Oh I was thinking there was a default ttl. I'll set a ttl on it for 3 days, since that should save it over a weekend if need be

Reply:
Does this rely on the prod redis server? I think @calellowitz work shows that we think we have a very high eviction rate of redis so I'd be hesitant to rely on anything definitely being there

Reply:
I guess it would. Are there likely to be followup efforts to fix that? It seems bad that one of our data stores wouldn't store data

Reply:
There are.

> It seems bad that one of our data stores wouldn't store data

I don't think we ever rely on redis absolutely having the data stored, so I wouldn't count it as a "data store", more of a caching layer that should have a real persistent backup method